test/test_integration.rb

```diff
@@ -239,6 +239,18 @@ def test_restart_closes_keepalive_sockets_workers
     assert_equal "Hello World", new_reply
   end
 
+  def test_restart_with_prune_bundler_keeps_bundle_gemfile_env
+    gemfile_path = File.expand_path("../Gemfile", __dir__)
+    duplicate_gemfile_path = gemfile_path + ".duplicate"
+    FileUtils.cp(gemfile_path, gemfile_path + ".duplicate")
```

Comment:
I'm not in love with the idea of doing a file copy in a test like this, but I thought this approach was preferable to having a duplicate Gemfile that is checked into git. I'm open to other suggestions though!

Reply:
I'm not sure I agree. I don't like having tests create/destroy files, especially in the root dir, as they tend to be created and then I ctrl-c out of my tests and the file is never deleted.

Why not just put a Gemfile somewhere appropriate in the test/ directory, like the other fixtures we have for Rackup files?

Reply:
It would remove ~3 lines from this test, at least.

Reply:
The Gemfile has to be a copy of the real Gemfile. If it is missing something or gets outdated, then Puma won’t start in the test. Also, because the Gemfile contains the directive `gemspec`, it must live in the root of the repo next to the puma.gemspec file.

Reply:
I do understand and agree with your sentiment about Ctrl-C, for sure!

Reply:
I have another idea for writing this test... I'll try to push another commit shortly. Thanks for looking!

Reply:
@nateberkopec please take another look at the PR. @bensie and I were able to make the test more generic by using PATH and `Bundler.with_original_env` rather than focusing solely on BUNDLER_GEMFILE. We believe this makes the intention clearer and the code simpler!